src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -3834,6 +3877,238 @@ impl Coordinator {
+                // large task for an operation we have to cover in tests anyway.
+                let source = catalog::Source::new(
+                    id,
+                    plan,
+                    // Use the same cluster ID.
+                    Some(cur_ingestion.instance_id),
+                    depends_on
+                        .into_iter()
+                        .chain(to_create_subsources.iter().map(|csp| csp.source_id))
+                        .collect(),
+                    cur_source.custom_logical_compaction_window,
+                    cur_source.is_retained_metrics_object,
+                );
+
+                // Get new ingestion description for storage.
+                let ingestion = match &source.data_source {
+                    DataSourceDesc::Ingestion(ingestion) => ingestion.clone(),
+                    _ => unreachable!("already verified of type ingestion"),
+                };
+
+                self.controller
+                    .storage
+                    .check_alter_collection(id, ingestion.clone())
+                    .map_err(|e| AdapterError::internal(ALTER_SOURCE, e))?;
+
+                let CreateSourceInner {
+                    mut ops,
+                    sources,
+                    if_not_exists_ids,
+                } = self
+                    .create_source_inner(session, to_create_subsources)
+                    .await?;
+
+                assert!(
+                    if_not_exists_ids.is_empty(),
+                    "IF NOT EXISTS not supported for ALTER SOURCE...ADD SUBSOURCES"
+                );
+
+                // Redefine source.
+                ops.push(Op::UpdateItem {
+                    id,
+                    // Look this up again so we don't have to hold an immutable reference to the
+                    // entry for so long.
+                    name: self.catalog.get_entry(&id).name().clone(),
+                    to_item: CatalogItem::Source(source),
+                });
+
+                self.catalog_transact(Some(session), ops).await?;
+
+                let mut source_ids = Vec::with_capacity(sources.len());
+                for (source_id, source) in sources {
+                    let source_status_collection_id =
+                        Some(self.catalog().resolve_builtin_storage_collection(
+                            &crate::catalog::builtin::MZ_SOURCE_STATUS_HISTORY,
+                        ));
+
+                    let (data_source, status_collection_id) = match source.data_source {
+                        // Subsources use source statuses.
+                        DataSourceDesc::Source => (DataSource::Other, source_status_collection_id),
+                        o => {
+                            unreachable!(
+                                "ALTER SOURCE...ADD SUBSOURCE only creates subsources but got {:?}",
+                                o
+                            )
+                        }
```

Comment:
Uncovered. Do you think it makes sense to cover this code with a test? Feel free to ignore if not.

Reply:
The top line (4073 in the diff this comment was on) should be covered, but those below it should not be.